pkg/api/validation/validation.go

```diff
@@ -1142,11 +1144,22 @@ func ValidatePodSpec(spec *api.PodSpec) errs.ValidationErrorList {
 // ValidatePodSecurityContext test that the specified PodSecurityContext has valid data.
 func ValidatePodSecurityContext(securityContext *api.PodSecurityContext, spec *api.PodSpec) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-
 	if securityContext != nil {
 		allErrs = append(allErrs, validateHostNetwork(securityContext.HostNetwork, spec.Containers).Prefix("hostNetwork")...)
 	}
-
+	if !validation.IsValidGroupId(*(securityContext.FSGroup)) {
```

Comment:
RunAsUser and FSGroup are both pointers so we should be doing a nil check to avoid a nil pointer dereference.

We should also add this check to the SecurityContext (container) validation if not already done for consistency.